### PR TITLE
enable llama2 awq

### PR DESCRIPTION
**Pull request by zhenwei-intel**
## Type of Change

feature

## Description

https://huggingface.co/TheBloke/Llama-2-7B-AWQ

```python
from transformers import AutoTokenizer, TextStreamer
from neural_speed import Model
model_name = "~/models/awq/Llama-2-7B-Chat-AWQ/"

prompt = "Once upon a time, a little girl"
tokenizer = AutoTokenizer.from_pretrained(model_name, trust_remote_code=True)
inputs = tokenizer(prompt, return_tensors="pt").input_ids
streamer = TextStreamer(tokenizer)

model = Model()
model.init(model_name, weight_dtype="int4", compute_dtype="int8", use_awq=True)
outputs = model.generate(inputs, streamer=streamer, max_new_tokens=300, do_sample=True)
```
outputs:
```bash
<s> Once upon a time, a little girl named Lily lived in a small village surrounded by rolling hills and vast fields of golden wheat.   
Lily was a curious child, always eager to explore the world around her. She spent her days wandering through the village, discovering n
ew things and making friends with the people she met along the way.                                                                    
One day, while wandering through the fields, Lily stumbled upon a hidden path that she had never seen before. The path was overgrown an
d looked as though it hadn't been used in years. Lily's curiosity was piqued, and she decided to follow the path to see where it led.  
As she walked along the path, Lily noticed that the air around her began to change. The scent of wildflowers filled the air, and she co
uld hear the chirping of birds in the distance. She felt as though she had entered a magical world, one that was hidden from the rest o
f the village.                                                                                                                         
Lily walked for hours, following the path through fields and forests, until she came upon a clearing surrounded by a ring of mushrooms.
 In the center of the clearing stood an ancient tree, its trunk twisted and gnarled with age.                                          
Suddenly, a figure emerged from the shadows cast by the tree. It was an old witch, her face lined with age and her eyes gleaming with a
 knowing light. She greeted L
```

## Expected Behavior & Potential Risk

the expected behavior that triggered by this PR 

## How has this PR been tested?

how to reproduce the test (including hardware information)

## Dependency Change?

any library dependency introduced or removed

### CODE REVIEW

File: neural_speed/__init__.py [hunk 1]
```diff
@@ -94,6 +94,10 @@ def init(self, model_name, not_quant=False, use_cache=False,
                 quant_desc += "_pc"
             else:
                 quant_desc += "_g{}".format(group_size)
+        if use_gptq:
+            quant_desc = "gptq"
+        if use_awq:
+            quant_desc = "awq"
```
Comment (airMeng) [inline neural_speed/__init__.py:100]: why not?

```suggestion
        if use_gptq:
            quant_desc += "_gptq"
        if use_awq:
            quant_desc += "_awq"
```

**Comment by airMeng** [comment]
I think AWQ should benefit accuracy, any number to prove it? Will MLPerf help?

File: neural_speed/__init__.py (hunk 1 above)
Comment (zhenwei-intel) [reply neural_speed/__init__.py:100]: quant_desc includes weight_dtype int 4/int 8, but for awq/gptq, we don't know the weight dtype from init

**Comment by zhenwei-intel** [comment]
> I think AWQ should benefit accuracy, any number to prove it? Will MLPerf help?

llama2 awq hf model: https://huggingface.co/TheBloke/Llama-2-7B-AWQ,

I will compute diff2 between cpp graph and hf awq

File: neural_speed/__init__.py (hunk 1 above)
Comment (airMeng) [reply neural_speed/__init__.py:100]: > quant_desc includes weight_dtype int 4/int 8, but for awq/gptq, we don't know the weight dtype from init

```awq.bin``` seems not meaningful enough, any other info to differ the model file?

**Review by a32543254 — APPROVED** [review]
LGTM

**Comment by a32543254** [comment]
Will we consider when insert mul op to graph during AWQ process?

**Comment by zhenwei-intel** [comment]
> Will we consider when insert mul op to graph during AWQ process?

mul is not in huggingface awq models, which is merged into layernorm.

**Comment by zhenwei-intel** [comment]
> > I think AWQ should benefit accuracy, any number to prove it? Will MLPerf help?
> 
> llama2 awq hf model: https://huggingface.co/TheBloke/Llama-2-7B-AWQ,
> 
> I will compute diff2 between cpp graph and hf awq

{'diff2': 0.049483024, 'cos': 0.9987812} close to hf awq

File: neural_speed/__init__.py (hunk 1 above)
Comment (zhenwei-intel) [reply neural_speed/__init__.py:100]: There is currently no convenient method, the file here is before convert, but info is available after convert

**Review by airMeng — APPROVED** [review]

**Pull request merged**